metadata/src/main/java/org/apache/kafka/controller/PartitionChangeBuilder.java

```diff
@@ -109,7 +111,8 @@ public PartitionChangeBuilder(
         int partitionId,
         IntPredicate isAcceptableLeader,
         MetadataVersion metadataVersion,
-        int minISR
+        int minISR,
+        Optional<QuorumControllerMetrics> metrics
```

Comment:
The metrics for unclean leader elections should be in ControllerMetadataMetrics, not here. See below.